### DISCUSSION TITLE
Invalid key in header: X-API-Key (intermittent error)

### DISCUSSION THREAD

**Discussion by AlexAS81** [OP-TED/eForms-SDK#574] [Q&A] [answered]
We have intermittent errors of type "Invalid key in header: X-API-Key" when calling the different operations (validate, submit,...) of the web services. What can it be due to? 

In the end it ends up working, the failures as I mentioned are intermittent.

Thanks in advance

**Comment by pcolast** [comment]
Good day,

Could you please provide the environment (PROD or Preview), HTTP error code and the approximate time when it happened ?

Thank you
Stefan

**Reply by AlexAS81** [reply]
Hello,

Environment: Preview
HTTP error code: 403
Time: 12/07/23 12:30

Body of the response:
{
	"errorCode": 403,
	"message": "Invalid key in header: X-API-Key",
	"timestamp": "2023-07-12T10:30:27Z"
}

Thank you,
Alex

**Reply by pcolast** [reply]
Hello Alex,

At the time you were making the tests there was a temporary disruption of one of the services.
There was an unforeseen intervention and therefore it was not communicated in advance.

We apologize for the inconvenience.

Have a nice day,
Stefan

**Reply by AlexAS81** [reply]
Hellos Stefan,

It happens to us almost every day. Are there unforeseen interventions often?

Have a nice day,
Alex

**Reply by pcolast** [reply, marked as answer]
Hello Alex,

We confirm that there is a general issue.
The investigation is ongoing.

We will confirm when it is resolved.

We apologize for the inconvenience.

Have a nice day,
Stefan

**Reply by pcolast** [reply]
Dear users, 

The issue was fixed in the Preview environment on 18/07.

Have a nice day,
Stefan

**Comment by rengit62** [comment]
I've got the same problem (in Preview) , sdk 1.7
Caused by: org.springframework.web.client.HttpClientErrorException$Forbidden: 403 : "{"errorCode":403,"message":"Invalid key in header: X-API-Key","timestamp":"2023-07-12T14:13:12Z"}"
        at org.springframework:5.3.27//org.springframework.web.client.HttpClientErrorException.create(HttpClientErrorException.java:109)
 
Thanks
Renato

**Reply by pcolast** [reply]
Hello Renato,

The timestamp is copied directly from the log, or you tried to adjust the UTC offset ?
Because you added the comment here at 14:59 UTC+2 which is 12:59 UTC and value of your timestamp "2023-07-12T14:13:12Z" is then in the future.

Thank you
Stefan

**Comment by rengit62** [comment]
Good Morning Stefan
I took the trace directly from the AS log. His time is 2 hours back CEST.
date now : Thu Jul 13 08:31:59 CEST 2023
Ciao
Renato

**Comment by rengit62** [comment]
Hi all !
from log now  ( Thu Jul 13 11:47:17 CEST 2023) : 

 - : org.springframework.web.client.HttpClientErrorException$Forbidden: 403 : "{"errorCode":403,"message":"Invalid key in header: X-API-Key","timestamp":"2023-07-13T09:40:29Z"}"

**Reply by pcolast** [reply]
Hello Renato,

Thank you for your input.
The issue is currently under investigation.

Have a nice day,
Stefan
